src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -59,7 +56,18 @@ private BoundNode RewriteTupleCreationExpression(BoundTupleCreationExpression no
                                                                                               TupleTypeSymbol.RestPosition - 1)
                                                                                       .Add(currentCreation);
 
-                        MethodSymbol constructor = tuple8Ctor.AsMember(underlyingTupleTypeChain.Pop());
+                        NamedTypeSymbol currentType = underlyingTupleTypeChain.Pop();
+                        var tuple8Ctor = (MethodSymbol)TupleTypeSymbol.GetWellKnownMemberInType(currentType.OriginalDefinition,
+                                                                                                TupleTypeSymbol.GetTupleCtor(TupleTypeSymbol.RestPosition),
+                                                                                                _compilation.Assembly,
+                                                                                                _diagnostics,
+                                                                                                node.Syntax);
+                        if ((object)tuple8Ctor == null)
+                        {
+                            return node;
+                        }
```

Comment:
I think we should still leave this part outside of the loop, because it is reasonable to require the same type used as an extension (the one with the Rest field) and we will enforce it as part of the shape check during metadata import. T